lib/php/lib/Thrift/Transport/TSocket.php

```diff
@@ -255,6 +255,16 @@ public function close()
   }
 
   /**
+   * Close the socket and try to
+   */
+  public function reconnect()
```

Comment:
Here we need a functionality to close a socket even if it`s persistent and open it again. This way we can react to the situation mentioned in the above comment.